### PR TITLE
Feat/1.14 relay reconstruc

### PR DESCRIPTION
**Pull request by DiyiWi**

### CODE REVIEW

File: internal/exchanger2/exchanger.go [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+	var (
+		srcAdaptName  string
+		destAdaptName string
+		serviceList   []string
+		err           error
+	)
+
+	if err := retry.Retry(func(attempt uint) error {
+		if srcAdaptName, err = ex.srcAdapt.Name(); err != nil {
+			ex.logger.Errorf("get name from srcAdapt", "error", err.Error())
```
Comment (jzhe886) [inline internal/exchanger2/exchanger.go:67]: return err?

File: internal/exchanger2/exchanger.go [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+		}
+		ex.destAdaptName = destAdaptName
+		return nil
+	}, strategy.Wait(3*time.Second)); err != nil {
+		return fmt.Errorf("retry error to get name from destAdapt: %w", err)
+	}
+
+	if err := retry.Retry(func(attempt uint) error {
+		if serviceList, err = ex.srcAdapt.GetServiceIDList(); err != nil {
+			ex.logger.Errorf("get serviceIdList from srcAdapt", "error", err.Error())
```
Comment (jzhe886) [inline internal/exchanger2/exchanger.go:86]: above

File: internal/exchanger2/exchanger.go [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+			ex.logger.Errorf("get name from srcAdapt", "error", err.Error())
+		}
+		ex.srcAdaptName = srcAdaptName
+		return nil
+	}, strategy.Wait(3*time.Second)); err != nil {
+		return fmt.Errorf("retry error to get name from srcAdapt: %w", err)
+	}
+	if err := retry.Retry(func(attempt uint) error {
+		if destAdaptName, err = ex.destAdapt.Name(); err != nil {
+			ex.logger.Errorf("get name from destAdapt", "error", err.Error())
```
Comment (jzhe886) [inline internal/exchanger2/exchanger.go:76]: above

File: internal/exchanger2/exchanger.go [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+}
+
+func (ex *Exchanger) listenIBTPFromDestAdapt() {
+	ch := ex.destAdapt.MonitorIBTP()
+	for {
+		select {
+		case <-ex.ctx.Done():
+			return
+		case ibtp, ok := <-ch:
+			ex.logger.Info("Receive ibtp from :", ex.destAdaptName)
```
Comment (jzhe886) [inline internal/exchanger2/exchanger.go:125]: more log

File: internal/exchanger2/exchanger.go [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+		}
+	}
+}
+
+func (ex *Exchanger) listenIBTPFromSrcAdapt() {
+	ch := ex.srcAdapt.MonitorIBTP()
+	for {
+		select {
+		case <-ex.ctx.Done():
+			return
```
Comment (jzhe886) [inline internal/exchanger2/exchanger.go:198]: exit log

File: internal/exchanger2/exchanger.go [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+		case ibtp, ok := <-ch:
+			ex.logger.Info("Receive interchain ibtp from :", ex.srcAdaptName)
```
Comment (jzhe886) [inline internal/exchanger2/exchanger.go:200]: more log

File: internal/exchanger2/exchanger.go (hunk 1 above)
Comment (DiyiWi) [reply internal/exchanger2/exchanger.go:67]: done

**Review by jzhe886 — APPROVED** [review]

**Review by 4ever9 — APPROVED** [review]

File: internal/adapt/appchain_adapter.go [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+
+}
+
+func (a *AppchainAdapter) init() error {
+	var err error
+
+	if err := retry.Retry(func(attempt uint) error {
+		a.client, a.pluginClient, err = plugins.CreateClient(a.config, nil)
+		if err != nil {
+			a.logger.Errorf("create client plugin", "error", err.Error())
```
Comment (4ever9) [inline internal/adapt/appchain_adapter.go:176]: return err?

**Review by jzhe886 — APPROVED** [review]

**Review by 4ever9 — APPROVED** [review]

**Pull request merged**